### PR TITLE
Comprehension internal tools/ rules analysis backend time filtering

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add time filtering params to backend of rules analysis

## WHY
so that Curriculum will be able to search through rules analysis based off of certain time parameters

## HOW
add optional time params passed to SQL/active record queries based off of `created_at` properties  for `Comprehension::Rule` and `FeedbackHistory`

### Screenshots
(If applicable. Also, please censor any sensitive data)

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/Filter-View-All-Sessions-by-time-frame-c36f841cec5845639d42c3a6af836e28

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  Yes
Have you deployed to Staging? | Not yet - deploying now!
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Design Review: If applicable, have you compared the coded design to the mockups? | N/A

### CODE REVIEW

File: services/QuillLMS/app/lib/rule_feedback_history.rb [hunk 1]
```diff
@@ -40,8 +44,10 @@ def self.feedback_history_to_json(f_h)
         }
     end
 
-    def self.generate_rulewise_report(rule_uid:, prompt_id: )
-        feedback_histories = FeedbackHistory.where(rule_uid: rule_uid, prompt_id: prompt_id, used: true)
+    def self.generate_rulewise_report(rule_uid:, prompt_id:, start_date:, end_date:)
+        start_date_param = start_date || FeedbackHistory.first&.created_at
```
Comment (happythenewsad) [inline services/QuillLMS/app/lib/rule_feedback_history.rb]: nit: you can avoid a DB call by just using a really early timestamp, like Jan 1 2000

**Review by happythenewsad — APPROVED** [review]

File: services/QuillLMS/app/controllers/api/v1/rule_feedback_histories_controller.rb [hunk 2]
```diff
@@ -3,7 +3,9 @@ def by_conjunction
         raise ArgumentError unless params.include?('activity_id') && params.include?('conjunction')
         report = RuleFeedbackHistory.generate_report(
             conjunction: params['conjunction'],
-            activity_id: params['activity_id']
+            activity_id: params['activity_id'],
+            start_date: params['start_date'],
+            end_date: params['end_date']
```
Comment (anathomical) [inline services/QuillLMS/app/controllers/api/v1/rule_feedback_histories_controller.rb]: You used `permit` on the other one of these, I think it's probably worth refactoring this controller to do the same (even though there's already clearly an existing pattern of not using `permit` that you were just following).  Same for the `rule_detail` action.

**Review by anathomical — APPROVED** [review]
This is probably fine as-is, but I think it'd be cool to use `permit` here somehow just for safety.

File: services/QuillLMS/app/controllers/api/v1/rule_feedback_histories_controller.rb (hunk 2 above)
Comment (eadams17) [reply services/QuillLMS/app/controllers/api/v1/rule_feedback_histories_controller.rb]: @anathomical good call, just refactored!

**Pull request merged**